core/src/epicli/data/common/ansible/playbooks/roles/prometheus/tasks/main.yml

```
@@ -11,11 +11,14 @@
   set_fact:
     does_k8s_exists: "(groups['kubernetes_master'] | default([])) | length > 0"
 
-- include: configure-k8s-apps-monitoring.yml
+- name: Configure kubernetes monitoring
+  when: does_k8s_exists
   become: true
+  include: configure-k8s-apps-monitoring.yml
+  environment:
+    KUBECONFIG: "{{ kubeconfig.local }}"
   tags:
     - kubernetes
```

```suggestion
- name: Configure Kubernetes monitoring
   when: does_k8s_exists
   include_tasks: configure-k8s-apps-monitoring.yml
   apply:
     environment:
       KUBECONFIG: "{{ kubeconfig.local }}"
     delegate_to: localhost
     become: false
     run_once: true
     tags:
        - kubernetes
  tags:
    - kubernetes
```
`delegate_to`, `become` & `run_once` not checked in this context, just idea to test.

---

There is not only `kubectl` inside, but also setting facts for example, so I suggest don't apply these changes

---

Updated my suggestion, kept only  `environment` and `include` -> `include_tasks`.  The `include` module is still not deprecated but it's a good idea to avoid using it, see [here](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/include_module.html#notes).

---

May be

```yaml
- name: Configure kubernetes monitoring
  when: does_k8s_exists
  include_tasks:
    file: configure-k8s-apps-monitoring.yml
    apply: 
      environment:
        KUBECONFIG: "{{ kubeconfig.local }}"
      tags:
        - kubernetes
  tags:
    - kubernetes
```